test/e2e/common_test.go

```diff
@@ -973,13 +973,25 @@ func populateCache(podman *PodmanTestIntegration) {
 // rmAll removes the directory and its content, when running rootless we use
 // podman unshare to prevent any subuid/gid problems
 func rmAll(podmanBin string, path string) {
+	// When using overlay, podman leaves a stray mount behind. This
+	// leak causes remote tests to take a loooooong time, and time out.
+	overlayPath := path + "/root/overlay"
+	if _, err := os.Stat(overlayPath); err == nil {
+		umount := exec.Command("umount", overlayPath)
+		umount.Stdout = GinkgoWriter
+		umount.Stderr = GinkgoWriter
+		if err = umount.Run(); err != nil {
+			GinkgoWriter.Printf("Error umounting %s: %v\n", overlayPath, err)
+		}
+	}
```

Comment:
This fixes the test hang.

We actually have a long history with this issue of leaked mounts: https://github.com/containers/buildah/blob/0d717cd0a52376c80cb527a344a1d7446b0d56e7/tests/helpers.bash#L116-L123

That dates to 2020. See https://github.com/containers/buildah/issues/1991 .

Is there really no way to fix this leak?